### ISSUE TITLE
Property testing

### ISSUE THREAD

**Issue by hackaugusto** [raiden-network/raiden#300]
Depends on: #299 

Property test raiden using [hyphotesis](https://pypi.python.org/pypi/hypothesis).

- Pogress in transfers must be made with valid messages only. Invalid ones must be ignored. Invalid messages include replay attacks, random data, completely invalid merkle root, merkle root missing locks, decreasing transferred_amount:
  - [ ] Mediated Transfer
  - [ ] Direct Transfer
  - [ ] Asset exchange
- Channels, test for: replay attacks, old messages, multiple calls, calls in the wrong order:
  - [ ] Openning
  - [ ] Closing
  - [ ] Settlement
  - [ ] Dispute

**Comment by LefterisJP** [comment]
@hackaugusto Can you confirm this is closed and if yes close the issue?

**Comment by hackaugusto** [comment]
we have property tests only for the smart contracts, I would like to have more tests for the protocol

**Comment by LefterisJP** [comment]
@hackaugusto Maybe would make sense to split it into smaller more specific issues then?

**Comment by hackaugusto** [comment]
Smart contract:

Channel Manager property testing #1154 
I did not add an issue for the end point registry, since that is temporary.
I did not add an issue for the registry smart contract since that is quite simple.

Initiator property testing #1155
Mediator property testing #1156
Target property testing #1157
At some point we also want to add issues for transfers / token swaps, but since that is not done, I didn't create the issues.
